examples/debugging/simple_spread/feedforward/decentralised/run_ippo_grid_search.py

```diff
@@ -0,0 +1,134 @@
+from mava.utils.loggers import logger_utils
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "env_name",
+    "simple_spread",
+    "Debugging environment name (str).",
+)
+flags.DEFINE_string(
+    "action_space",
+    "discrete",
+    "Environment action space type (str).",
+)
+
+flags.DEFINE_string(
+    "mava_id",
+    str(datetime.now()),
+    "Experiment identifier that can be used to continue experiments.",
+)
+flags.DEFINE_string("base_dir", "~/mava", "Base dir to store experiments.")
+
+
+def create_and_run_lp_program(config: Dict) -> None:
+    """Code to run lp program using config.
+
+    Args:
+        config : hyperparam config.
+    """
+    # Environment.
+    environment_factory = functools.partial(
+        debugging_utils.make_environment,
+        env_name=FLAGS.env_name,
+        action_space=FLAGS.action_space,
+    )
+
+    # Networks.
+    def network_factory(*args: Any, **kwargs: Any) -> Any:
+        return ippo.make_default_networks(  # type: ignore
+            policy_layer_sizes=(256, 256, 256),
+            critic_layer_sizes=(512, 512, 256),
+            *args,
+            **kwargs,
+        )
+
+    # Used for checkpoints, tensorboard logging and env monitoring
+    experiment_path = f"{FLAGS.base_dir}/{FLAGS.mava_id}"
+
+    # Log every [log_every] seconds.
+    log_every = 10
+    logger_factory = functools.partial(
+        logger_utils.make_logger,
+        directory=FLAGS.base_dir,
+        to_terminal=True,
+        to_tensorboard=True,
+        time_stamp=FLAGS.mava_id,
+        time_delta=log_every,
+    )
+
+    # Optimisers.
+    policy_optimiser = optax.chain(
+        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+    )
+
+    critic_optimiser = optax.chain(
+        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+    )
+
+    # Create the system.
+    system = ippo.IPPOSystem()
+
+    # Build the system.
+    system.build(
+        environment_factory=environment_factory,
+        network_factory=network_factory,
+        logger_factory=logger_factory,
+        experiment_path=experiment_path,
+        policy_optimiser=policy_optimiser,
+        critic_optimiser=critic_optimiser,
+        run_evaluator=True,
+        sample_batch_size=5,
+        num_epochs=config["num_epochs"],
+        num_executors=1,
+        multi_process=True,
+        clip_value=False,
+        termination_condition={"executor_steps": 10000},
+    )
+    # Launch the system.
+    worker_manager = system.launch()
+    # Wait for worker to finish - this critical to waiting for program to finish.
+    if worker_manager:
+        worker_manager.wait()
+
+
+def main(_: Any) -> None:
+    """Runs lp inside a new thread.
+
+    Args:
+        _ : unused.
+    """
+    config = {}
+    # Loop through a hyperparam such as num_epochs
+    for num_epochs in [5, 10, 15]:
+        config["num_epochs"] = num_epochs
+        create_and_run_lp_program(config)
```

Comment:
Probably @KaleabTessera can answer you about that

Reply:
@DriesSmit Outside of tensorboad this isn't an issue since we log the config for each run. In tf, I think we should log hyperparams/config as opposed to changing the id since we could have multiple configs params that we sweep. I will create an issue for this.

Reply:
Sounds good :slightly_smiling_face: Are we logging the the mava_id for each run as well? Then I'm happy :+1: